### ISSUE TITLE
ID is None in model observer message when model objects are deleted with atomic transactions

### ISSUE THREAD

**Issue by dkmorgan** [NilCoalescing/djangochannelsrestframework#207]
I have a ModelViewSet with a custom `destroy()` definition decorated with `@transaction.atomic`.
When the model object is deleted, the `message` passed into the model_observer handler does not contain the ID.

I have checked that the ID is present in the instance when `ModelObserver.post_delete_receiver()` is called.
However, the ID has changed to None when `ModelObserver.post_change_receiver()` is called.

The ID does not change to None if I remove the `@transaction.atomic` decorator from `destroy()` in the ModelViewSet.

Is this behavior expected?
If so, are there any workarounds to retain the ID when model objects are deleted via atomic transactions?
I would like to avoid defining post_delete signal handlers for all my models to send the delete notifications via the AsyncAPIConsumer containing the model_observer definition.

**Comment by hishnash** [comment]
Soory I did not spot this sooner.

This is interesting, are other fields still populated (is it just the ID that is lost?)

**Comment by dkmorgan** [comment]
Not exactly sure, but #212 may have resolved this issue?

**Comment by hishnash** [comment]
@dkmorgan  yes you are correct, deleted models should now send a update even including the pk of the item that was deleted as the message is created before the transaction completes (but sent after).

**Comment by dkmorgan** [comment]
@hishnash 

Thank you for the confirmation.
Apologies for the change in subject, but are there any plans of an official release?
It has been quite a while since the last one.

**Comment by hishnash** [comment]
@dkmorgan  yes there are 2 other PRs I want to merge, and some fixes to documentation.
